ecl/hql/hqlgram.y

```diff
@@ -5986,6 +5987,23 @@ primexpr1
                             parser->normalizeExpression($7, type_int, false);
                             $$.setExpr(createValue(no_regex_find, subType.getLink(), $3.getExpr(), $5.getExpr(), $7.getExpr(), $8.getExpr()));
                         }
+    | REGEXFINDSET '(' expression ',' expression regexOpt ')'
+                        {
+                            parser->normalizeExpression($3, type_stringorunicode, false);
+                            Owned<ITypeInfo> retType;
+                            if(isUnicodeType($3.queryExprType()))
+                            {
+                                retType.setown(makeUnicodeType(UNKNOWN_LENGTH, 0));
+                                parser->normalizeExpression($5, type_unicode, false);
+                            }
+                            else
+                            {
+                                parser->normalizeExpression($5, type_string, false);
+                                retType.setown(makeStringType(UNKNOWN_LENGTH));
+                            }
+
+                            $$.setExpr(createValue(no_regex_findset, makeSetType(retType.getLink()), $3.getExpr(), $5.getExpr(), $6.getExpr()));
```

Comment:
minor: Also good to add a $1 parameter to the setExpr() call to ensure the position is set correctly.